web/packages/core/src/ruffle-player.ts

```diff
@@ -677,6 +685,26 @@ export class RufflePlayer extends HTMLElement {
         }
     }
 
+    /**
+     * Called when entering / leaving fullscreen on non-Safari
+     */
+    private fullScreenChange(): void {
+        // check if we have just stopped being full screen
+        if (document.fullscreenElement === null) {
```

Comment:
Sorry for the delay. I'm worried this doesn't cover cases where the user forcibly enters fullscreen (e.g. typing `player.requestFullscreen()` in DevTools). We can change this to a two-way update:
```ts
this.instance?.set_fullscreen(document.fullscreenElement !== null);
```

Reply:
Not if we combine this and the Safari one.

The possibilities are:

|  | Safari | Other |
| --- | --- | --- |
| fullscreenElement | undefined | null or object |
| webkitFullscreenElement | null or object | undefined |

so if fullscreenElement isn't null, either we have a full screen element or we're in Safari.

You can try something like:
```
this.instance?.set_fullscreen((document.fullscreenElement !== null && document.webkitFullscreenElement === undefined) || (document.webkitFullscreenElement !== null && document.fullscreenElement === undefined));
```

but I think the code deduplication is outweighed by the additional complexity.

I'd prefer to do this change, but not deduplicate. What do you think?

Reply:
This could be simplified based on the fact that `null` and `undefined` are falsy:
```ts
this.instance?.set_fullscreen(document.fullscreenElement || document.webkitFullscreenElement);
```
So if this is doable I still prefer deduplicating the methods.

Reply:
That's nice: seems like it should work for all cases and looks simple too. Thanks.